packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,286 @@
+
+  await step('Should be checked', async () => {
+    await userEvent.click(firstCheckbox)
+    expect(firstCheckboxInput).toBeChecked()
+  })
+
+  await step('Should not be checked', async () => {
+    await userEvent.click(secondCheckbox)
+    expect(secondCheckboxInput).not.toBeChecked()
+  })
+}
+
+export const Immediate = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    [true]
+    <va-form immediate>
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+    [false]
+    <va-form>
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+  `,
+})
+
+Immediate.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const [firstInput, secondInput] = canvas.getAllByRole('textbox', { name: '' }) as HTMLElement[]
+
+  await step('Displays error message', async () => {
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('true')
+  })
+
+  await step('Does not display error message', async () => {
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+}
+
+export const HideErrorMessages = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    [true]
+    <va-form
+      role
+      immediate
+      hideErrorMessages
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+    [false]
+    <va-form
+      role
+      immediate
+      :hideErrorMessages="false"
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+  `,
+})
+
+HideErrorMessages.play = async ({ step }) => {
+  const [firstDefinedInput, secondDefineInput] = document.querySelectorAll('[role] > div > div:nth-child(2)')
+
+  await step('Does not display validation error', async () => {
+    expect(firstDefinedInput).toBeDefined()
+  })
+
+  await step('Displays validation error', async () => {
+    expect(secondDefineInput).toBeUndefined()
+  })
+}
+
+export const HideErrors = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    [true]
+    <va-form
+      role
+      immediate
+      hideErrors
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+    [false]
+    <va-form
+      role
+      immediate
+      :hideErrors="false"
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+  `,
+})
+
+HideErrors.play = async ({ step }) => {
+  const [firstInput, secondInput] = document.querySelectorAll('[role] > div')
+
+  await step('Does not highlight invalid input', async () => {
+    expect(firstInput).not.toHaveClass('va-input-wrapper--error')
+  })
+
+  await step('Highlights invalid input', async () => {
+    expect(secondInput).toHaveClass('va-input-wrapper--error')
+  })
+}
+
+export const Focus = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input />
```

Comment:
For Focus and FocusInvalid we need 2 inputs in template.

If feature doesn't work properly - focus might select not first input.
For focusInvalid - it might be indifferent to input valid state - and we won't know from story.